### ISSUE TITLE
Layout of elements

### ISSUE THREAD

**Issue by brgrz** [dominicbarnes/jWizard#34]
Hey Dominic, I think it makes more sense to render the step title next to a step, for example if steps are fieldsets just render the legend out (don't move title out of content).

On the other hand, I think it'd be better if menu was rendered outside of content.

Essentially it'd be nice to have all the elements standalone.

**Comment by brgrz** [comment]
of course I could start working on it but wanted to check it out with you first, what you think

**Comment by dominicbarnes** [comment]
Yeah, the next step for jWizard was to make the structure of the wizard really flexible. I haven't been sure exactly how to do this, without making awkward/complex configuration objects. Earlier version of jWizard had options like `counter.location` that made configuration bulky and awkward so I'd rather not presume to know every layout permutation my user's would try to use.

My last big idea was to allow jWizard to recognize elements with particular class names as particular pieces of the interface. The jQuery UI Slider does this now, as you can put custom handle elements inside the root element as long as they have a specific class. That is a pretty simple case, so I'm not sure if it really applies here.

My other big idea is to break this down into even more pieces, (this is where I begin to prefer Component.js over jQuery UI) rather than requiring jWizard to manage everything. I would have several widgets (like Step, Menu, Counter, etc) that the developer would piece together depending on their needs. This would necessarilly require a lot more up-front work, but huge amount of flexibility is the payoff.

Any thoughts?

**Comment by brgrz** [comment]
IMHO The switch to component.io comes with a steeper learning and integration curve for most, while having a few additional options to turn stuff on and off (not much, I guess 3-5 will suffice) is much simpler.

**Comment by dominicbarnes** [comment]
Yeah, it would likely become an entirely new project if I were to rewrite jWizard using component. I have found component to be really powerful (after surmounting the curve of course) where I work and all my future work will likely be done that way. (if not in Angular.js, which I've also found to be really powerful)
